### ISSUE TITLE
VM failed for unhandled reason, Query <EVM.Query: fetch contract ... >

### ISSUE THREAD

**Issue by agroce** [crytic/echidna#180]
https://solidity.readthedocs.io/en/v0.5.3/solidity-by-example.html

```
pragma solidity >=0.4.22 <0.6.0;

contract SimpleAuction {
    // Parameters of the auction. Times are either
    // absolute unix timestamps (seconds since 1970-01-01)
    // or time periods in seconds.
    address payable public beneficiary;
    uint public auctionEndTime;

    // Current state of the auction.
    address public highestBidder;
    uint public highestBid;

    // Allowed withdrawals of previous bids
    mapping(address => uint) pendingReturns;

    // Set to true at the end, disallows any change.
    // By default initialized to `false`.
    bool ended;

    // Events that will be emitted on changes.
    event HighestBidIncreased(address bidder, uint amount);
    event AuctionEnded(address winner, uint amount);

    // The following is a so-called natspec comment,
    // recognizable by the three slashes.
    // It will be shown when the user is asked to
    // confirm a transaction.

    /// Create a simple auction with `_biddingTime`
    /// seconds bidding time on behalf of the
    /// beneficiary address `_beneficiary`.
    constructor(
        uint _biddingTime,
        address payable _beneficiary
    ) public {
        beneficiary = _beneficiary;
        auctionEndTime = now + _biddingTime;
    }

    /// Bid on the auction with the value sent
    /// together with this transaction.
    /// The value will only be refunded if the
    /// auction is not won.
    function bid() public payable {
        // No arguments are necessary, all
        // information is already part of
        // the transaction. The keyword payable
        // is required for the function to
        // be able to receive Ether.

        // Revert the call if the bidding
        // period is over.
        require(
            now <= auctionEndTime,
            "Auction already ended."
        );

        // If the bid is not higher, send the
        // money back.
        require(
            msg.value > highestBid,
            "There already is a higher bid."
        );

        if (highestBid != 0) {
            // Sending back the money by simply using
            // highestBidder.send(highestBid) is a security risk
            // because it could execute an untrusted contract.
            // It is always safer to let the recipients
            // withdraw their money themselves.
            pendingReturns[highestBidder] += highestBid;
        }
        highestBidder = msg.sender;
        highestBid = msg.value;
        emit HighestBidIncreased(msg.sender, msg.value);
    }

    /// Withdraw a bid that was overbid.
    function withdraw() public returns (bool) {
        uint amount = pendingReturns[msg.sender];
        if (amount > 0) {
            // It is important to set this to zero because the recipient
            // can call this function again as part of the receiving call
            // before `send` returns.
            pendingReturns[msg.sender] = 0;

            if (!msg.sender.send(amount)) {
                // No need to call throw here, just reset the amount owing
                pendingReturns[msg.sender] = amount;
                return false;
            }
        }
        return true;
    }

    /// End the auction and send the highest bid
    /// to the beneficiary.
    function auctionEnd() public {
        // It is a good guideline to structure functions that interact
        // with other contracts (i.e. they call functions or send Ether)
        // into three phases:
        // 1. checking conditions
        // 2. performing actions (potentially changing conditions)
        // 3. interacting with other contracts
        // If these phases are mixed up, the other contract could call
        // back into the current contract and modify the state or cause
        // effects (ether payout) to be performed multiple times.
        // If functions called internally include interaction with external
        // contracts, they also have to be considered interaction with
        // external contracts.

        // 1. Conditions
        require(now >= auctionEndTime, "Auction not yet ended.");
        require(!ended, "auctionEnd has already been called.");

        // 2. Effects
        ended = true;
        emit AuctionEnded(highestBidder, highestBid);

        // 3. Interaction
        beneficiary.transfer(highestBid);
    }

    function echidna_test () external returns (bool) {
    	return true;
    }
}
```

No need for config, just `echidna-test SimpleAuction.sol` and you'll see the hang between 10-50 iterations (usually about 10).  My solc is:

`Version: 0.5.4+commit.9549d8ff.Darwin.appleclang`

**Comment by agroce** [comment]
Looks like it causes an internal error, really.  Here's a shrunk version that causes the issue:

```
echidna-test Shrunk.sol --config shrink.yaml 
Shrunk.sol:1:1: Warning: Source file does not specify required compiler version! Consider adding "pragma solidity ^0.5.4;"
contract SimpleAuction {
^ (Relevant source part starts here and spans across multiple lines).
Analyzing contract: Shrunk.sol:SimpleAuction
echidna-test: VM failed for unhandled reason, Query <EVM.Query: fetch contract 0000000000000000000000000000000000000000>. This shouldn't happen. Please file a ticket with this error message and steps to reproduce!
```

shrink.yaml:
```
testLimit: 100
dashboard: false
```

Shrunk.sol:
```
contract SimpleAuction {
    address payable public beneficiary;
    uint public auctionEndTime;
    uint public highestBid;
    mapping(address => uint) pendingReturns;
    constructor(
    ) public {
    }
    function bid() public payable {
        require(
            now <= auctionEndTime,
            "Auction already ended."
        );
        require(
            msg.value > highestBid,
            "There already is a higher bid."
        );
        if (highestBid != 0) {
        }
    }
    function withdraw() public returns (bool) {
        uint amount = pendingReturns[msg.sender];
        if (amount > 0) {
            if (!msg.sender.send(amount)) {
            }
        }
    }
    function auctionEnd() public {
        beneficiary.transfer(highestBid);
    }
    function echidna_test () external returns (bool) {
    }
}
```

**Comment by agroce** [comment]
Shorter yet:

```
contract SimpleAuction {
    address payable public beneficiary;
    uint public highestBid;
    function auctionEnd() public {
        beneficiary.transfer(highestBid);
    }
    function echidna_test () external returns (bool) {
    	return true;
    }
}
```

**Comment by agroce** [comment]
Is this just the known issue?  Hanging vs. the error seems user-unfriendly, though, not sure if that's generic to the problem or not..

**Comment by JackHFeng** [comment]
I am getting the same issue, were you able to figure out why?

**Comment by ggrieco-tob** [comment]
This is a known issue caused by a `Query` error in HEVM when echidna tries to interact with addresses that are not defined. In this case: `beneficiary.transfer(highestBid)` (which is going to be `0x0` in this example). While we fix this issue, please comment any line sending ether or calling non-existing contracts.

**Comment by JackHFeng** [comment]
> This is a known issue caused by a `Query` error in HEVM when echidna tries to interact with addresses that are not defined. In this case: `beneficiary.transfer(highestBid)` (which is going to be `0x0` in this example). While we fix this issue, please comment any line sending ether or calling non-existing contracts.

Thank you so much!

**Comment by ggrieco-tob** [comment]
This should be solved in #212 but please test it. :smile:
